packages/okta-auth-js/lib/browser/browser.js

```diff
@@ -162,8 +169,16 @@ proto.features.isTokenVerifySupported = function() {
   return typeof crypto !== 'undefined' && crypto.subtle && typeof Uint8Array !== 'undefined';
 };
 
+proto.features.hasTextEncoder = function() {
+  return typeof TextEncoder !== 'undefined';
+};
+
 proto.features.isPKCESupported = function() {
-  return proto.features.isTokenVerifySupported();
+  return proto.features.isTokenVerifySupported() && proto.features.hasTextEncoder();
+};
+
+proto.features.isHTTPS = function() {
+  return window.location.protocol === 'https:';
```

Comment:
Should this allow for running on localhost for development purposes?

Reply:
This function is only used in reporting errors, so the isPKCESupported() check already failed.  

Depending on your browser, the secure webcrypto check will or will not pass on an HTTP localhost url, so if you see this message, the browser will NOT allow HTTP, even for localhost, and there's not much we can do about that.

Reply:
@whawker In all browsers I have tested, PKCE will work on `http://localhost`. (IE does require a TextEncoder polyfill). In most browsers PKCE will fail on HTTP with any origin other than localhost, since `crypto.subtle` requires a "secure context". Firefox is the exception which seems to allow HTTP domains and `crypto.subtle`.